packages/core-plugin-api/src/apis/definitions/AnalyticsApi.ts

```diff
@@ -0,0 +1,120 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ApiRef, createApiRef } from '../system';
+import { AnalyticsDomainValue } from '../../analytics/types';
+
+/**
+ * Represents an event worth tracking in an analytics system that could inform
+ * how users of a Backstage instance are using its features.
+ *
+ * Note that attributes about the Backstage user or about the plugin tracking
+ * the event are inferred and captured separately on the Analytics Domain and
+ * do not need to be passed on the event itself.
+ */
+export type AnalyticsEvent = {
+  /**
+   * A string that identifies the event being tracked by the type of action the
+   * event represents. Examples include:
+   *
+   * - view
+   * - click
+   * - filter
+   * - search
+   * - hover
+   * - scroll
+   */
+  verb: string;
+
+  /**
+   * A string that uniquely identifies the object that the verb or action is
+   * being taken on. Examples include:
+   *
+   * - The path of the page viewed
+   * - The url of the link clicked
+   * - The value that was filtered by
+   * - The text that was searched for
+   */
+  noun: string;
+
+  /**
+   * An optional numeric value relevant to the event that could be aggregated
+   * by analytics tools. Examples include:
+   *
+   * - The index or position of the clicked element in an ordered list
+   * - The percentage of an element that has been scrolled through
+   * - The amount of time that has elapsed since a fixed point
+   */
+  value?: number;
+
+  /**
+   * Optional context with any additional dimensions or metrics that could be
+   * forwarded on to analytics systems.
+   */
+  context?: AnalyticsEventContext;
+};
+
+/**
+ * An analytics event combined with domain attributes.
+ */
+export type DomainDecoratedAnalyticsEvent = AnalyticsEvent & {
+  /**
+   * Domain metadata relating to where the event was captured and by whom. This
+   * could include information about the route, plugin, or component in which
+   * an event was captured.
+   */
+  domain: AnalyticsDomainValue;
+};
+
+/**
+ * A structure allowing other arbitrary metadata to be provided by analytics
+ * event emitters.
+ */
+export type AnalyticsEventContext = {
+  [attribute in string]: string | boolean | number;
+};
+
+/**
+ * Represents a tracker with methods that can be called to track events in a
+ * configured analytics service.
+ */
+export type AnalyticsTracker = {
+  captureEvent: (
+    verb: string,
+    noun: string,
+    value?: number,
+    context?: AnalyticsEventContext,
```

Comment:
Perhaps the last two could be merged into an options object as they're not used as often?

Reply:
Yeah, I think that's probably a good idea.